src/main.rs

```diff
@@ -101,7 +115,15 @@ fn read_can(pub_path: &str, can_interface: &str) -> JoinHandle<u32> {
  */
 fn read_siren(pub_path: &str, send_map: Arc<RwLock<HashMap<u32, EncodeData>>>) -> JoinHandle<()> {
     let mut client = MqttClient::new(pub_path, "calypso-encoder");
-    client.connect().expect("Could not connect to Siren!");
+    // client.connect().expect("Could not connect to Siren!");
+
+    while !client.is_connected() {
+        println!("[read_siren] Unable to connect to Siren, going into reconnection mode.");
+        if client.reconnect().is_ok() {
+            println!("[read_siren] Reconnected to Siren!");
+        }
+    }
+
```

Comment:
I think this code is necessary.  However, if you remove it, will the reconnection block down below (line 172), be enough to maintain a connection just by waiting until failure?  Im guessing `client.start_consumer()` would fail as well, but may be good to test.

Reply:
start_consumer may crash